test/unit/gem_typo_test.rb

```diff
@@ -1,51 +1,52 @@
-      gem_typo = GemTypo.new("four")
-      assert_equal false, gem_typo.protected_typo?
-    end
+    deleted = create(:rubygem, name: "deleted_job_active_record")
+    create(:version, rubygem: deleted, created_at: Time.now.utc, indexed: false)
+  end
 
-    should "return false for gem name size below protected threshold" do
-      gem_typo = GemTypo.new("fou")
-      assert_equal false, gem_typo.protected_typo?
-    end
+  should "return false for exact match" do
+    gem_typo = GemTypo.new("delayed_job_active_record")
+    refute gem_typo.protected_typo?
+  end
 
-    context "size equals protected threshold" do
-      should "return true for one character distance" do
-        gem_typo = GemTypo.new("fous")
-        assert_equal true, gem_typo.protected_typo?
-      end
+  should "return false for any exact match so that owner of the delayed_job_active_record Gem can push an update with existing typosquat" do
+    existing_typo = build(:rubygem, name: "delayed-job-active-record")
+    existing_typo.save(validate: false)
+    create(:version, rubygem: existing_typo, created_at: Time.now.utc)
 
-      should "return false for two character distance" do
-        gem_typo = GemTypo.new("foss")
-        assert_equal false, gem_typo.protected_typo?
-      end
-    end
+    gem_typo = GemTypo.new("delayed_job_active_record")
+    refute gem_typo.protected_typo?
+  end
+
+  should "return false for an exact match of a yanked gem so a gem with an identical name can be published in the future" do
+    gem_typo = GemTypo.new("deleted_job_active_record")
+    refute gem_typo.protected_typo?
+  end
 
-    context "size above protected threshold" do
-      should "return true for two character distance" do
-        gem_typo = GemTypo.new("fourss")
-        assert_equal true, gem_typo.protected_typo?
-      end
+  should "return false for a underscore variation match of a yanked gem so a gem with a similar name can be published in the future" do
+    gem_typo = GemTypo.new("deleted-job-active_record")
+    refute gem_typo.protected_typo?
+  end
 
-      should "return false for exceptions" do
-        create(:gem_typo_exception, name: "fourss")
+  context "typo squat on an existing Gem name" do
+    should "return true for one -/_ character change" do
+      gem_typo = GemTypo.new("delayed-job_active_record")
+      assert gem_typo.protected_typo?
+    end
 
-        gem_typo = GemTypo.new("fourss")
-        assert_equal false, gem_typo.protected_typo?
-      end
+    should "return true for two -/_ change" do
+      gem_typo = GemTypo.new("delayed-job_active-record")
+      assert gem_typo.protected_typo?
+    end
 
-      should "return false for three characher distance" do
-        gem_typo = GemTypo.new("foursss")
-        assert_equal false, gem_typo.protected_typo?
-      end
+    should "return true for three -/_ character change" do
+      gem_typo = GemTypo.new("delayed-job-active-record")
+      assert gem_typo.protected_typo?
```

Comment:
Please add a test case for `delayed-job-activerecord` being protected.

Reply:
Added on branch, including additional tests for missing -/_.